hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -153,180 +192,128 @@ def collect_current_houdini_session(self, settings, parent_item):
-            tk_mantra_nodes = mantranode_app.get_nodes()
-        except AttributeError, e:
-            self.logger.warning(
-                "Unable to query the session for tk-houdini-mantranode "
-                "instances. It looks like perhaps an older version of the "
-                "app is in use which does not support querying the nodes. "
-                "Consider updating the app to allow publishing their outputs."
-            )
-            return
-
-        # retrieve the work file template defined by the app. we'll set this
-        # on the collected alembicnode items for use during publishing.
-        work_template = mantranode_app.get_work_file_template()
-
-        for node in tk_mantra_nodes:
-
-            out_path = mantranode_app.get_output_path(node)
-
-            if not os.path.exists(out_path):
-                continue
-
-            self.logger.info(
-                "Processing sgtk_mantra node: %s" % (node.path(),))
-
-            # allow the base class to collect and create the item. it
-            # should know how to handle the output path
-            item = super(HoudiniSessionCollector, self)._collect_file(
-                parent_item,
-                out_path,
-                frame_sequence=True
-            )
-
-            # the item has been created. update the display name to
-            # include the node path to make it clear to the user how it
-            # was collected within the current session.
-            item.name = "%s (%s)" % (item.name, node.path())
-
-            if work_template:
-                item.properties["work_template"] = work_template
-
-            self._mantra_nodes_collected = True
+        logger = publisher.logger
+        logger.debug("Getting publish name for path: %s ..." % (path,))
+
+        path_info = publisher.util.get_file_path_components(path)
+        filename = path_info["filename"]
+
+        version_pattern_match = re.search(r"(.*)([._-])v(\d+)\.?(.*)$", filename, re.IGNORECASE)
+
+        publish_name = filename
+        if version_pattern_match:
+            # found a version number, use the other groups to remove it
+            prefix = version_pattern_match.group(1)
+            extension = version_pattern_match.group(4) or ""
+            if extension:
+                publish_name = "%s.%s" % (prefix, extension)
+            else:
+                publish_name = prefix
+
+        frame_pattern_match = re.search(r"(.*)([._-])(\d+)\.(.*)$", publish_name, re.IGNORECASE)        
+        if frame_pattern_match and sequence:
+            # found a frame number, replace it with #s
+            prefix = frame_pattern_match.group(1)
+            frame_sep = frame_pattern_match.group(2)
+            frame = frame_pattern_match.group(3)
+            display_str = "#" * len(frame)
+            extension = frame_pattern_match.group(4) or ""
+            publish_name = "%s%s%s.%s" % (
+                prefix, frame_sep, display_str, extension)
```

Comment:
You could try using named regex elements, i.e.

```python
        frame_pattern_match = re.search(
            r"(?P<prefix>.*)"
            r"(?P<separator>[._-])"
            r"(?P<frame>\d+)"
            r"(?P<extension>\..*)$",
            publish_name,
        )
        if frame_pattern_match and sequence:
            # found a frame number, replace it with #s
            elements = frame_pattern_match.groupdict()
            elements['frame'] = "#" * len(elements['frame'])
            publish_name = "{prefix}{separator}{frame}{extension}"
            publish_name = publish_name.format(**elements)
```

Also, since you're taking out the `.` before the extension during the match, then adding it back in again, you might be better to just including it in the original regular expression

Reply:
this isn't needed anymore